### PR TITLE
fix publishing sha256 for release build

### PR DESCRIPTION
**Pull request by rkpagadala**
fix script to publish sha256
add link to sha256 for deb file

### CODE REVIEW

**Comment by hklai** [comment]
Has this been tested?

And shellcheck is failing

File: release/cloud_builder.sh [hunk 1]
```diff
@@ -111,7 +111,9 @@ fi
 if [ "${BUILD_DEBIAN}" == "true" ]; then
   mkdir -p "${OUTPUT_PATH}/deb"
   cp        "${ISTIO_OUT}/istio-sidecar.deb"   "${OUTPUT_PATH}/deb/"
-  sha256sum "${ISTIO_OUT}/istio-sidecar.deb" > "${OUTPUT_PATH}/deb/istio-sidecar.deb.sha256"
+  pushd "${OUTPUT_PATH}/deb"
+    sha256sum "istio-sidecar.deb" > "istio-sidecar.deb.sha256"
+  popd
 fi
```
Comment (stewart-yu) [inline release/cloud_builder.sh]: i hesitate about here change. seems change code struct, need?

Comment (rkpagadala) [reply release/cloud_builder.sh]: previously the code would generate the following file
a6b5ab210923606b6b3fc83a102c60b2ef08270e90b7e6ce83ef13be43168c76  /workspace/go/out/linux_amd64/release/istio-sidecar.deb
now it generates the following file
a6b5ab210923606b6b3fc83a102c60b2ef08270e90b7e6ce83ef13be43168c76  istio-sidecar.deb

having the directory path is incorrect

**Comment by rkpagadala** [comment]
@hklai fixed https://github.com/rkpagadala/istio/releases/tag/untagged-1fa985001cf21293864c

File: release/cloud_builder.sh (hunk 1 above)
Comment (hklai) [reply release/cloud_builder.sh]: Not sure I follow. 

Is a6b5ab210923606b6b3fc83a102c60b2ef08270e90b7e6ce83ef13be43168c76 istio-sidecar.deb one file?

**Comment by hklai** [comment]
I can't see https://github.com/rkpagadala/istio/releases/tag/untagged-1fa985001cf21293864c

Maybe it's not published and it's in your own org?

**Comment by rkpagadala** [comment]
@hklai I assumed you could see draft releases there, apparently not. I changed it to an actual release https://github.com/rkpagadala/istio/releases/tag/1.1.0.snapshot.1

File: release/cloud_builder.sh (hunk 1 above)
Comment (rkpagadala) [reply release/cloud_builder.sh]: This was the file previously generated 
https://storage.googleapis.com/istio-release/releases/1.1.0.snapshot.1/deb/istio-sidecar.deb.sha256
inside it there is one line of the form 
<sha256> <filename with path>
after the change the file has
<sha256> <filename without path>

**Pull request merged**